Adding example for cloud-events

adam-intel opened the pull request.

jim-wang-yutsung commented on app-services/cloud-event-transforms/README.md:
```diff
@@ -0,0 +1,9 @@
+
+In this example we start off by using the included `transforms.NewConversion().TransformToCloudEvent` function to transform an Edgex Event into a cloud-event.  Then in the following app function we send the cloud-event, then transform it back to an Edgex Event using `transforms.NewConversion().TransformFromCloudEvent` and print out the result.  
+
+You can trigger this this pipeline by using the http trigger using a tool like postman and sending an edgex event. 
```
```suggestion
You can trigger this pipeline by using the http trigger from a tool like postman and sending an edgex event in the JSON body as payload. 
```

jim-wang-yutsung commented on app-services/cloud-event-transforms/main.go:
```diff
@@ -0,0 +1,149 @@
+	ctx = cloudevents.ContextWithHeader(ctx, "demo", "header value")
+	c, err := cloudevents.NewDefaultClient()
+	if err != nil {
+		edgexcontext.LoggingClient.Error(fmt.Sprintf("failed to create client, %v", err))
```
Shouldn't we return false here?

adam-intel replied:
Yes I'll add that

jim-wang-yutsung commented on app-services/cloud-event-transforms/main.go:
```diff
@@ -0,0 +1,149 @@
+	}
+	for _, cloudevent := range events {
+		if _, resp, err := c.Send(ctx, cloudevent); err != nil {
+			edgexcontext.LoggingClient.Error(fmt.Sprintf("failed to send: %v", err))
```
return false for this error case?

adam-intel replied:
Yes I'll add that

jim-wang-yutsung commented on app-services/cloud-event-transforms/pkg/transforms/conversion.go:
```diff
@@ -0,0 +1,128 @@
+		return false, errors.New("No Event Received")
+	}
+	edgexcontext.LoggingClient.Debug("Transforming from CloudEvent to models.Event")
+	if result, ok := params[0].([]cloudevents.Event); ok {
```
do the opposite to return error earlier if cast is NOT ok so that the whole code logic needs not indent.

jim-wang-yutsung reviewed (changes requested):
Some suggestions to change...

adam-intel committed:
App Service examples of cloudevent transforms

This pr creates an app function examples of transforming edgex events cloudevents and back

closes: #255 and #254

Signed-off-by: adam-w-mitchell <adam.w.mitchell@intel.com>

jim-wang-yutsung reviewed (approved):
LGTM

rsdmike committed:
Merge branch 'master' into sample-cloudevent
